termwiz/src/lineedit/mod.rs

```diff
@@ -811,7 +811,6 @@ impl<'term> LineEditor<'term> {
     }
 
     fn read_line_impl(&mut self, host: &mut dyn LineEditorHost) -> Result<Option<String>> {
-        self.line.clear();
```

Comment:
ah, I didn't spot this initially. I think this is a big semantic change that breaks behavior. You can probably see this in the debug overlay (ctrl-shift-L) if you type in `foo` and hit enter.  Does it cause foo to show up in the next line? 

What I think should be done here is to adjust this method:

```rust
fn read_line_impl(&mut self,
    host: &mut dyn LineEditorHost,
    initial_value: Option<&str>
) -> Result<Option<String>> {
    self.line.clear();
    if let Some(value) = initial_value {
        self.line.set_line_and_cursor(value, value.len());
    }
```

Then, refactor:

```rust
pub fn read_line(&mut self, host: &mut dyn LineEditorHost) -> Result<Option<String>>
```

 into

```rust
pub fn read_line_with_optional_initial_value(&mut self, host: &mut dyn LineEditorHost, initial_value: Option<&str>) -> Result<Option<String>>
```

 that calls the above function:

```rust
...
let res = self.read_line_impl(host, initial_value);
...
```

Then finally add back the original method, but make it call the refactored one:

```rust
pub fn read_line(&mut self, host: &mut dyn LineEditorHost) -> Result<Option<String>> {
   self.read_line_with_optional_initial_value(host, None)
}
```

Note that the public methods on this struct need to have doc comments because they are part of the public API that is published to https://docs.rs/termwiz/latest/termwiz/lineedit/struct.LineEditor.html

Reply:
> I think this is a big semantic change that breaks behavior. You can probably see this in the debug overlay (ctrl-shift-L) if you type in foo and hit enter. Does it cause foo to show up in the next line?

From what I can test, it does not appear as if removal of this line causes old contents to appear anywhere; I tested various combinations of launching REPL, custom editor, pressing enter, cancelling input with Esc and Ctrl+C and other things.

If I understand correctly, in every instance where a `LineEditor` struct is spawned, it's `read_line` method is only ever called once, with no loops. As such, the "clear" function does not actually do anything: it only ever sees an already empty, freshly created line.

Reply:
The termwiz crate is a public export and used by others, so just looking at the usage within wezterm is not sufficient to determine that nothing is impacted by this change.

I went ahead and merged this with my suggested changes.

Thank you!

Reply:
This makes sense. Thank you for reviewing this